src/OpenTelemetry.Sampler.AWS/AWSXRayRemoteSamplerBuilder.cs

```diff
@@ -66,8 +67,17 @@ public AWSXRayRemoteSamplerBuilder SetEndpoint(string endpoint)
     /// <summary>
     /// Returns a <see cref="AWSXRayRemoteSampler"/> with configuration of this builder.
     /// </summary>
-    /// <returns>an instance of <see cref="AWSXRayRemoteSampler"/>.</returns>
-    public AWSXRayRemoteSampler Build()
+    /// <returns>an instance of <see cref="Trace.Sampler"/>.</returns>
+    public Trace.Sampler Build()
+    {
+        using var rootSampler = this.BuildXraySampler();
+        return new ParentBasedSampler(rootSampler);
+    }
+
+    // This is intended for testing to check that the XRayRemoteSampler is built with the correct attributes
+    // Should not be exposed to public as the public build method should return the sampler wrapped inside
+    // ParentBasedSampler.
+    internal AWSXRayRemoteSampler BuildXraySampler()
```

Comment:
Is there a way to avoid needing a special constructor just for testing?

Could the tests call `Build`, get the `ParentBasedSampler` and then access the `AWSXrayRemoteSampler` from that class some way?

Reply:
Using reflection allows us to get the private sampler in the ParentBasedSampler class. However doing so is kindoff a code smell since at this point we are relying on the internal variable naming of ParentBasedSampler and if that changes, the test breaks. I would prefer using the special constructor in this case since it's being used in the actual public facing Build function and can be used to test the AWSXrayRemoteSampler being built properly.